### DISCUSSION TITLE
Best way to have hoverable text

### DISCUSSION THREAD

**Discussion by FFY00** [Textualize/textual#69] [Q&A]
Hey, I would like to use textual to write an interactive version of hexdump, where it would show the data on the left and show the data details on the right when you hover it. What would be the best way to achieve this?
I think probably the best way would be having a widget for the text and a way to locate the cursor while rendering? I am not sure about the performance impact of that, but an event based approach could also be possible.

What do you think? Would something like this be in scope?

**Comment by willmcgugan** [comment]
Definitely in scope. You can have overlapping layers by setting the z value and you can move things around by setting `layout_offset. In theory that is enough to get "tooltips" or similar.

The API there is still rough around the edges. But it will eventually be a first class feature.
